lib/tbot/service_spiffe_workload_api.go

```diff
@@ -702,30 +617,27 @@ func (s *SPIFFEWorkloadAPIService) FetchX509Bundles(
 	ctx := srv.Context()
 	s.log.InfoContext(ctx, "FetchX509Bundles stream opened by workload")
 	defer s.log.InfoContext(ctx, "FetchX509Bundles stream has closed")
-	renewCh, unsubscribe := s.trustBundleBroadcast.subscribe()
-	defer unsubscribe()
+
+	bundleSetCh, stopBundleSetCh := s.trustBundleCache.Subscribe()
+	defer stopBundleSetCh()
 
 	for {
+		var bundleSet *spiffe.BundleSet
+		select {
+		case <-ctx.Done():
+			return nil
+		case bundleSet = <-bundleSetCh:
+		}
+		// TODO: If no value is available within the first X seconds, should we drop
```

Comment:
@timothyb89 curious on your thoughts here - the client may well have it's own timeouts that would kick in, and, regardless it'll reconnect if the connection is dropped.

Reply:
Hmm, I do like the idea of failing fast. I'd maybe lean toward trying that and seeing if anything breaks / we get complaints?